frontend/src/scenes/insights/utils/queryUtils.ts

```diff
@@ -41,14 +41,19 @@ export const compareQuery = (a: Node, b: Node, opts?: CompareQueryOpts): boolean
         const { source: sourceA, ...restA } = a
         const { source: sourceB, ...restB } = b
         return (
-            objectsEqual(objectCleanWithEmpty(restA), objectCleanWithEmpty(restB)) &&
-            compareDataNodeQuery(sourceA, sourceB, opts)
+            objectsEqual(
+                objectCleanWithEmpty(removeUndefinedAndNull(restA)),
```

Comment:
fly-by: objectClean already removes undefined values. Why not add an optional parameter that removes nulls as well, instead of iterating the object twice and having disjoint naming for the cleaning functions.

Also I think there were some edge cases with filters where missing and nulls weren't actually the same thing. I think this is UI only when adding new filters and it might be ok here.

Reply:
Ah, so `objectCleanWithEmpty` doesn't recurse through the object, I tried to modify it to do this, but I spent over an hour trying to debug why it wasn't returning what I thought it should - I knew this other util existed, worked, and is recursive, so I cut my losses and went for that instead - I do agree though, `objectCleanWithEmpty` _should_ do this